gcc/rust/backend/rust-compile-resolve-path.cc

```diff
@@ -32,32 +32,39 @@
 namespace Rust {
 namespace Compile {
 
-void
-ResolvePathRef::visit (HIR::QualifiedPathInExpression &expr)
+template <typename T>
+tree
+ResolvePathRef::resolve_path_like (T &expr)
 {
-  auto final_segment = HIR::PathIdentSegment::create_error ();
   if (expr.is_lang_item ())
-    final_segment
-      = HIR::PathIdentSegment (LangItem::ToString (expr.get_lang_item ()));
+    {
+      auto lang_item
+	= Analysis::Mappings::get ().get_lang_item_node (expr.get_lang_item ());
+
+      // FIXME: Is that correct? :/
+      auto final_segment
```

Comment:
yes this should work the only reason i passed the node around was because it was handy to get the location and node id so in theory it can fall back to the query system interface.

Not part of your pr but yeah this has given me alot of though of loads of good-first-pr's to cleanup the interfaces here
its super confusing for new people.